net/shadowsocks-libev/Makefile

```diff
@@ -118,6 +118,9 @@ CONFIGURE_ARGS += \
 	--disable-assert \
 	--disable-ssp \
 
+TARGET_CFLAGS += -flto
+TARGET_LDFLAGS += -Wl,--gc-sections,--as-needed
```

Comment:
I am not sure here.  It should be

```
TARGET_CFLAGS += -ffunction-sections -fdata-sections
TARGET_LDFLAGS += -Wl,--gc-sections
```

compiler flag `-flto` and linker flag `--as-needed` should be not relevant here.  Please correct me if it's otherwise.

Reply:
Yes, at least using --gc-sections.
@dengqf6 
Does enabling LTO help bringing the binary size down?

Reply:
@diizzyy 
Yes, it is the LTO that does the main job here (shrinks ipk size by 20%). It works very well with certain programs such as aria2 (#7569), but it may increase size or cause problems (#7853)  in some cases.
`-Wl,--gc-sections,--as-needed` reduces size further by a few hundred bytes

Reply:
Interesting, looks like the compiler is smart enough when handling `-flto` and `-Os`.

The time cost is negligible.  I think we should apply both.

```
EXTRA_CFLAGS += -ffunction-sections -fdata-sections -flto
EXTRA_LDFLAGS += -Wl,--gc-sections -flto
```

original

```
-rwxr-xr-x 1 yunion yunion 468532 Jun  9 07:50 ss-local
-rwxr-xr-x 1 yunion yunion 199072 Jun  9 07:50 ss-manager
-rwxr-xr-x 1 yunion yunion 408432 Jun  9 07:50 ss-redir
-rwxr-xr-x 1 yunion yunion 514408 Jun  9 07:50 ss-server
-rwxr-xr-x 1 yunion yunion 396332 Jun  9 07:50 ss-tunnel
```


-flto

```
-rwxr-xr-x 1 yunion yunion 381320 Jun  9 07:59 ss-local
-rwxr-xr-x 1 yunion yunion 159256 Jun  9 07:59 ss-manager
-rwxr-xr-x 1 yunion yunion 326548 Jun  9 07:59 ss-redir
-rwxr-xr-x 1 yunion yunion 426012 Jun  9 07:59 ss-server
-rwxr-xr-x 1 yunion yunion 314800 Jun  9 07:59 ss-tunnel
```

compiler flags `-fxx-sections` with linker flag `-Wl,--gc-sections`

```
-rwxr-xr-x 1 yunion yunion 429740 Jun  9 08:01 ss-local
-rwxr-xr-x 1 yunion yunion 162672 Jun  9 08:01 ss-manager
-rwxr-xr-x 1 yunion yunion 371196 Jun  9 08:01 ss-redir
-rwxr-xr-x 1 yunion yunion 465484 Jun  9 08:01 ss-server
-rwxr-xr-x 1 yunion yunion 359076 Jun  9 08:01 ss-tunnel
```

both

```
-rwxr-xr-x 1 yunion yunion 380476 Jun  9 08:03 ss-local
-rwxr-xr-x 1 yunion yunion 158828 Jun  9 08:03 ss-manager
-rwxr-xr-x 1 yunion yunion 325860 Jun  9 08:03 ss-redir
-rwxr-xr-x 1 yunion yunion 425088 Jun  9 08:03 ss-server
-rwxr-xr-x 1 yunion yunion 314112 Jun  9 08:03 ss-tunnel
```

Reply:
With `-Wl,--as-needed` was specified, we can tweak out unneeded dependencies for some components.  It should save a lot more bytes!

```
--- aa  2019-06-10 01:58:55.162898922 +0000
+++ ab  2019-06-10 01:56:52.664821524 +0000
@@ -1,41 +1,32 @@
 ## ipkg-install/usr/bin/ss-local
 libev.so.4
-libcares.so.2
 libsodium.so.23
 libmbedcrypto.so.3
 libpcre.so.1
 libgcc_s.so.1
 libc.so
 ## ipkg-install/usr/bin/ss-manager
 libev.so.4
-libcares.so.2
-libsodium.so.23
-libmbedcrypto.so.3
-libpcre.so.1
 libgcc_s.so.1
 libc.so
 ## ipkg-install/usr/bin/ss-nat
 ## ipkg-install/usr/bin/ss-redir
 libev.so.4
-libcares.so.2
 libsodium.so.23
 libmbedcrypto.so.3
-libpcre.so.1
 libgcc_s.so.1
 libc.so
 ## ipkg-install/usr/bin/ss-server
 libev.so.4
 libcares.so.2
 libsodium.so.23
 libmbedcrypto.so.3
 libpcre.so.1
 libgcc_s.so.1
 libc.so
 ## ipkg-install/usr/bin/ss-tunnel
 libev.so.4
-libcares.so.2
 libsodium.so.23
 libmbedcrypto.so.3
-libpcre.so.1
 libgcc_s.so.1
 libc.so
```

Reply:
Thanks for the info, I have updated the commit to remove unnecessary dependencies